### PR TITLE
Enable click handling, as well as hover, on QuestionButton

### PR DESCRIPTION
**Pull request by azangru**
## Type
- Improvement to existing feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1343

## Description
- Add click handling to `QuestionButton`. A click on `QuestionButton` will show the tooltip; the next click will hide it, and so on.
- Also, keep the previous behaviour on hover (wait for a while and show the tooltip)
- Update some CSS to prevent text selection and other nuisance on repeat clicks

## Deployment URL
http://question-button.review.ensembl.org

### CODE REVIEW

File: src/ensembl/src/shared/components/question-button/QuestionButton.tsx [hunk 1]
```diff
@@ -14,10 +14,10 @@
  * limitations under the License.
  */
 
-import React, { useState, useEffect } from 'react';
+import React from 'react';
 import classNames from 'classnames';
 
-import useHover from 'src/shared/hooks/useHover';
+import { useQuestionButtonBehavior } from './useQuestionButtonBehavior';
```
Comment (imransl) [inline src/ensembl/src/shared/components/question-button/QuestionButton.tsx:20]: Aren't we sticking with UK spellings for variables?

File: src/ensembl/src/shared/components/question-button/useQuestionButtonBehavior.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useReducer, useEffect } from 'react';
+import useHover from 'src/shared/hooks/useHover';
```
Comment (imransl) [inline src/ensembl/src/shared/components/question-button/useQuestionButtonBehavior.tsx:18]: Separate these imports?

**Review by msmanoj — APPROVED** [review]

File: src/ensembl/src/shared/components/question-button/QuestionButton.tsx (hunk 1 above)
Comment (jyothishnt) [reply src/ensembl/src/shared/components/question-button/QuestionButton.tsx:20]: we may need to change this too https://github.com/Ensembl/ensembl-client/blob/dev/src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item/ProteinsListItem.tsx#L102 😆

Comment (jyothishnt) [reply src/ensembl/src/shared/components/question-button/QuestionButton.tsx:20]: I just did a quick check in the code and found we are sticking on to British English. Sorry @azangru  :-)

Comment (azangru) [reply src/ensembl/src/shared/components/question-button/QuestionButton.tsx:20]: We can't change that, Jyo; it's the browser API, and browsers speak American English:

https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView

This is why I would love us to use American English in code. All the other examples of the word `behaviour` in our code are for display labels; which is totally fine, we could write labels in Chinese hyerogliphs for all we care. But the code itself would be much more consistent if it stuck to American spelling 😞

Comment (jyothishnt) [reply src/ensembl/src/shared/components/question-button/QuestionButton.tsx:20]: oh oh!

**Comment by jyothishnt** [comment]
When I click to show tooltip, should it close only when I click outside?
It closes when I hover the icon for few seconds and move away. Just checking if this is the correct behaviour.

File: src/ensembl/src/shared/components/question-button/QuestionButton.test.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      );
+      const questionButton = container.querySelector(
+        '.questionButton'
+      ) as HTMLElement;
+
+      userEvent.hover(questionButton);
+      expect(questionButton.querySelector('.pointerBox')).toBeFalsy();
+
+      act(() => {
+        jest.advanceTimersByTime(TOOLTIP_TIMEOUT + 10);
```
Comment (jyothishnt) [inline src/ensembl/src/shared/components/question-button/QuestionButton.test.tsx:45]: TOOLTIP_TIMEOUT = 800  Is it in milliseconds? If so, does it wait for 8 seconds here when we run this test?

File: src/ensembl/src/shared/components/question-button/QuestionButton.tsx (hunk 1 above)
Comment (azangru) [reply src/ensembl/src/shared/components/question-button/QuestionButton.tsx]: It is with heavy heart that I have changed behavior for behaviour 😞 

✅

**Comment by azangru** [comment]
> Currently if the tooltip is activated by click and we do a hover action (hover in and out), it closes the tooltip.

Thanks for spotting this, @jyothishnt . I updated the code to prevent this from happening, and added a test to confirm that this is now the case.

**Review by jyothishnt — APPROVED** [review]

**Pull request merged**